docs/source/compression/compression_evaluator.rst

```diff
@@ -0,0 +1,305 @@
+
+* The ``model`` is a wrapped model from the original model, it has a similar structure to the model to be pruned, so it can share training function with the original model.
+* ``optimizers`` are re-initialized according to ``optimizers`` passed to the evaluator and the wrapped model's parameters.
+* ``criterion`` also based on the ``criterion`` passed to the evaluator, it might be modified in some algorithms.
+* If users use ``lr_schedulers`` in the ``training_func``, NNI will re-initialize the ``lr_schedulers`` with the re-initialized optimizers.
+* ``max_steps`` is the NNI training duration limitation. An integer means that after ``max_steps`` steps, the training should stop. ``None`` means NNI doesn't limit the duration, it is up to users to decide when to stop.
+* ``max_epochs`` is similar to the ``max_steps``, it controls the longest training epochs.
+
+Note that ``optimizers`` and ``lr_schedulers`` passed to the ``training_func`` have the same type as the ``optimizers`` and ``lr_schedulers`` passed to evaluator,
+a single ``torch.optim.Optimzier``/ ``torch.optim._LRScheduler`` instance or a list of them.
+
+optimizers
+^^^^^^^^^^
+A single traced optimizer instance or a list of traced optimizers by ``nni.trace``.
+
+NNI may modify the ``torch.optim.Optimizer`` member function ``step`` and/or optimize compressed models,
+so NNI needs to have the ability to re-initialize the optimizer. ``nni.trace`` can record the initialization parameters of a function/class,
+which can then be used by NNI to re-initialize the optimizer for a new but structurally similar model.
+
+criterion
+^^^^^^^^^
+``criterion`` function is used to compute the loss in the training function, the inputs of it are ``input`` and ``target``.
+Sometimes, NNI needs to add additional loss to sparse the model parameters. NNI will change the ``criterion`` to do that,
+e.g. in the ``training_func``, ``loss = criterion(input, target)`` will change to ``loss = patched_criterion(input, target)``.
+
+lr_schedulers
+^^^^^^^^^^^^^
+A single traced lr_scheduler instance or a list of traced lr_schedulers by ``nni.trace``.
+
+For the same reason with ``optimizers``, NNI needs the traced lr_scheduler to re-initialize it.
+
+dummy_input
+^^^^^^^^^^^
+``dummy_input`` is used to trace the model graph, it's same with ``example_inputs`` in `torch.jit.trace <https://pytorch.org/docs/stable/generated/torch.jit.trace.html?highlight=torch%20jit%20trace#torch.jit.trace>`_.
+It's only used by scheduled pruner intermediate model speedup right now.
+
+evaluating_func
+^^^^^^^^^^^^^^^
+This is the function used to evaluate the compressed model performance.
+The input is a model and the output is a float metric or a dict.
+NNI will take the float metric as the model score, and assume the higher score means the better performance.
+If you want to provide additional information, please put it into a dict and NNI will take the value of key ``default`` as evaluation metric.
+
```

Comment:
it is also worth to mention that not all the arguments of TorchEvaluator must be provided. Some pruners only require evaluate_func as they do not train the model, some pruners only require training_func. Please refer to each pruner's (or quantizer's) doc to check the required arguments. But, it is fine to provide more arguments than the pruner's need.

Reply:
add them to docstring `Notes`